### PR TITLE
Jay/docs/ansible

### LINKED ISSUE #4987: Improve Ansible Documentation
## Summary
We need to improve documentation [here](https://goteleport.com/teleport/docs/admin-guide/#ansible-integration) regarding Ansible. 

> Ansible uses the OpenSSH client by default. This makes it compatible with Teleport without any extra work, except configuring OpenSSH client to work with Teleport Proxy

But how?

We should provide two examples:
A) Configuring ~/.ssh/config to proxy through the Teleport Proxy. 
B) Configuring your Ansible inventory to proxy through the Teleport Proxy.

### Audience
Users of Ansible 

### Location
Existing guide. https://goteleport.com/teleport/docs/admin-guide/#ansible-integration

### PR DESCRIPTION
**Pull request by yjperez**
Expanding on Ansible Integration documentation located in the Admin Guide. Included general setup information which was assumed in the original documentation.

### CODE REVIEW

File: docs/5.0/admin-guide.md [hunk 1]
```diff
@@ -1268,15 +1268,52 @@ $ tctl status
...
-```
+There are two recommended scenarios:
+
+1. Ansible connects directly to Teleport Node using built in ssh library over port 3022 and Teleport user credentials
+
+2. Ansible connects to Teleport Node using the Teleport Proxy as a jumphost. This method uses Teleport user credentials over port 3022
+
+The configuration for openssh is as follows:
+
+1. Add hosts to know hosts in Ansible node
```
Comment (a-palchikov) [inline docs/5.0/admin-guide.md:1282]:
```suggestion
1. Add hosts to known hosts in Ansible node
```

File: docs/5.0/admin-guide.md [hunk 2]
```diff
@@ -1268,15 +1268,52 @@ $ tctl status
...
+    ```bash
+    $ eval `ssh-agent`
+    $ tsh --proxy=root.example.com login
+    # Verify the certs were added
+    $ ssh-add -L
+    ```
+
+!!! tip "NOTE"
+
+    Additional Teleport and OpenSSH intergration information can be located [HERE](https://goteleport.com/teleport/docs/openssh-teleport/#using-openssh-client)
```
Comment (a-palchikov) [inline docs/5.0/admin-guide.md:1315]:
```suggestion
    Additional Teleport and OpenSSH integration information can be located [HERE](https://goteleport.com/teleport/docs/openssh-teleport/#using-openssh-client)
```

File: docs/5.0/admin-guide.md [hunk 3]
```diff
@@ -1268,15 +1268,52 @@ $ tctl status
...
-```bash
-scp_if_ssh = True
-```
+There are two recommended scenarios:
+
+1. Ansible connects directly to Teleport Node using built in ssh library over port 3022 and Teleport user credentials
+
+2. Ansible connects to Teleport Node using the Teleport Proxy as a jumphost. This method uses Teleport user credentials over port 3022
+
+The configuration for openssh is as follows:
```
Comment (a-palchikov) [inline docs/5.0/admin-guide.md:1280]: Use `OpenSSH` consistently?

File: docs/5.0/admin-guide.md [hunk 4]
```diff
@@ -1268,15 +1268,52 @@ $ tctl status
...
+
+1. Add hosts to know hosts in Ansible node
+    * From Teleport Auth Node:
+        ```bash
+        $ tsh auth export --type=host
+        ```
+    * Add the output on Ansible node `/.ssh/known_hosts` by modifying the node section to match the nodes you are reaching. (Note: wildcards are accepted)
```
Comment (a-palchikov) [inline docs/5.0/admin-guide.md:1287]:
```suggestion
    * Add the output on Ansible node `/.ssh/known_hosts` by modifying the node section to match the nodes you are reaching (note: wildcards are accepted).
```

File: docs/5.0/admin-guide.md [hunk 5]
```diff
@@ -1268,15 +1268,52 @@ $ tctl status
...
+2. Ansible connects to Teleport Node using the Teleport Proxy as a jumphost. This method uses Teleport user credentials over port 3022
+
+The configuration for openssh is as follows:
+
+1. Add hosts to known hosts in Ansible node
+    * From Teleport Auth Node:
+        ```bash
+        $ tsh auth export --type=host
+        ```
+    * Add the output on Ansible node `/.ssh/known_hosts` by modifying the node section to match the nodes you are reaching (note: wildcards are accepted).
```
Comment (travelton) [inline docs/5.0/admin-guide.md:1287]:
```suggestion
    * Add the output on Ansible node `/.ssh/known_hosts` by modifying the host name to match the nodes you are connecting to (note: wildcards are accepted).
```
The OpenSSH documentation calls this a host name, not a node. Suggest using the OpenSSH documentation nomenclature for alignment.

File: docs/5.0/admin-guide.md [hunk 6]
```diff
@@ -1268,15 +1268,52 @@ $ tctl status
...
+    ```bash
+    @cert-authority <NODE(S)> ssh-rsa AAA...REDACTED...TD9 logins=-teleport-nologin-f0d...REDACTED...fb1&type=host
```
Comment (travelton) [inline docs/5.0/admin-guide.md:1289]:
```suggestion
    @cert-authority <hostname.example.com|hostname|10.0.0.0> ssh-rsa AAA...REDACTED...TD9 logins=-teleport-nologin-f0d...REDACTED...fb1&type=host
```
Maybe this to clarify what's suggested in the previous sentence.

File: docs/5.0/admin-guide.md [hunk 7]
```diff
@@ -1268,15 +1268,52 @@ $ tctl status
...
-Teleport Proxy:
+Teleport without any extra work, except configuring OpenSSH client to work with Teleport Proxy.
 
-* configure your OpenSSH to connect to Teleport proxy and use `ssh-agent` socket
-* enable scp mode in the Ansible config file (default is `/etc/ansible/ansible.cfg` ):
 
-```bash
-scp_if_ssh = True
-```
+There are two recommended scenarios:
```
Comment (travelton) [inline docs/5.0/admin-guide.md:1274]: I would recommend organizing this differently. 

There are two methods for OpenSSH clients to **connect** to Teleport Nodes; connecting to a Teleport Node directly over port 3022, or proxying through the Teleport Proxy over port 3023. 

There are two methods for OpenSSH clients to **authenticate** to Teleport Nodes; using Teleport credentials from a `tsh` session, or using a long lived SSH certificate exported from the Teleport Auth server.

File: docs/5.0/admin-guide.md [hunk 8]
```diff
@@ -1268,15 +1268,52 @@ $ tctl status
...
+        $ tctl auth sign --user=<teleport-user> --format=openssh --out=<file-name>
+        ```
+    * Extract both files created and add to `/.ssh/` on Ansible node
+
+1. Enable `scp mode` in the Ansible config file (default is `/etc/ansible/ansible.cfg` ):
+        ``` bash
+        scp_if_ssh = True
+        ```
+
+1. Configure your OpenSSH to connect to Teleport proxy and use `ssh-agent` socket:
```
Comment (travelton) [inline docs/5.0/admin-guide.md:1305]:
```suggestion
1. Configure your OpenSSH client to connect to Teleport proxy and use `ssh-agent` socket:
```

File: docs/5.0/admin-guide.md [hunk 9]
```diff
@@ -1268,15 +1268,52 @@ $ tctl status
...
+    * Modify ssh config file to include the proxy node and the desired Teleport nodes with `ProxyCommand` or `ProxyJump` parameters configured
+
+1. Create identity file and extract to Ansible node
+    * From Teleport Auth Node:
+        ```bash
+        $ tctl auth sign --user=<teleport-user> --format=openssh --out=<file-name>
+        ```
+    * Extract both files created and add to `/.ssh/` on Ansible node
+
+1. Enable `scp mode` in the Ansible config file (default is `/etc/ansible/ansible.cfg` ):
```
Comment (travelton) [inline docs/5.0/admin-guide.md:1300]: This is common between all connection and authentication types, so only mention it once, after you've given examples connection and authentication types.

File: docs/5.0/admin-guide.md [hunk 10]
```diff
@@ -1268,15 +1268,52 @@ $ tctl status
...
+    ```bash
+    $ eval `ssh-agent`
+    $ tsh --proxy=root.example.com login
+    # Verify the certs were added
+    $ ssh-add -L
+    ```
+
+!!! tip "NOTE"
+
+    Additional Teleport and OpenSSH integration information can be located [HERE](https://goteleport.com/teleport/docs/openssh-teleport/#using-openssh-client)
```
Comment (travelton) [inline docs/5.0/admin-guide.md:1315]:
```suggestion
    Additional Teleport and OpenSSH integration information can be located [here](https://goteleport.com/teleport/docs/openssh-teleport/#using-openssh-client)
```

File: docs/5.0/admin-guide.md [hunk 11]
```diff
@@ -1268,15 +1268,52 @@ $ tctl status
...
+        ```bash
+        $ tsh auth export --type=host
+        ```
+    * Add the output on Ansible node `/.ssh/known_hosts` by modifying the node section to match the nodes you are reaching (note: wildcards are accepted).
+    ```bash
+    @cert-authority <NODE(S)> ssh-rsa AAA...REDACTED...TD9 logins=-teleport-nologin-f0d...REDACTED...fb1&type=host
+    ```
+    * Modify ssh config file to include the proxy node and the desired Teleport nodes with `ProxyCommand` or `ProxyJump` parameters configured
+
+1. Create identity file and extract to Ansible node
```
Comment (travelton) [inline docs/5.0/admin-guide.md:1293]: Given the above comment about restructuring the layout, I think this section will need to be updated to reflect the options. 

Examples:

Connection Type (include very simple and minimal ~/.ssh/config examples)
- Direct to Node
- Proxy to Node

Authentication Type
- `tsh` credentials
- `tctl auth sign` long lived SSH creds

**Review by travelton — CHANGES_REQUESTED** [review]
Good start! Just need to a bit of structure adjustments.

File: docs/5.0/admin-guide.md (hunk 3 above)
Comment (yjperez) [reply docs/5.0/admin-guide.md:1280]: yes, will look over and make OpenSSH consistent across the document

**Comment by russjones** [comment]
@travelton @yjperez What's the status of this PR? If we're not going to make these changes in the short term, let's close and reopen later?

**Comment by travelton** [comment]
Let's close this for now. Our docs have changed, so we need to move these changes over to that new platform.

**Pull request closed without merging**